source/Missions/Services/Agents/Messages/AgentMovement.cs

```diff
@@ -0,0 +1,185 @@
+        public Vec2 AgentMovementDirection;
+
+        /// <summary>
+        /// The <see cref="AgentEquipmentData"/> of the <see cref="Agent"/>.
+        /// </summary>
+        [ProtoMember(7)]
+        public AgentEquipmentData AgentEquipmentData;
+
+        /// <summary>
+        /// Guid of the associated <see cref="Agent"/>.
+        /// </summary>
+        [ProtoMember(8)]
+        public Guid AgentId { get; }
+
+        /// <summary>
+        /// The <see cref="PacketType"/>.
+        /// </summary>
+        public PacketType PacketType => PacketType.Movement;
+
+        /// <summary>
+        /// The <see cref="DeliveryMethod"/>.
+        /// </summary>
+        public DeliveryMethod DeliveryMethod => DeliveryMethod.Unreliable;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="agent"></param>
+        /// <param name="guid"></param>
+        public AgentMovement(
+            Vec3 agentCurrentPosition, 
+            Vec2 agentMovementDirection, 
+            AgentEquipmentData agentEquipmentData, 
+            AgentActionData agentActionData,
+            AgentMountData agentMountData, 
+            Guid guid)
+        {
+            AgentId = guid;
+            CurrentPosition = agentCurrentPosition;
+            AgentMovementDirection = agentMovementDirection;
+            AgentEquipmentData = agentEquipmentData;
+            ActionDataDelta = agentActionData;
+            MountDataDelta = agentMountData;
+        }
+
+        /// <summary>
+        /// Re-calculate this <see cref="AgentMovement"/>.
+        /// </summary>
+        /// <param name="change"></param>
+        public void CalculateMovement(LookDirectionChanged change)
+        {
+            LookDirectionDelta = change.LookDirection;
+        }
+
+        /// <summary>
+        /// Re-calculate this <see cref="AgentMovement"/>.
+        /// </summary>
+        /// <param name="change"></param>
+        public void CalculateMovement(MovementInputVectorChanged change)
+        {
+            InputDirectionDelta = change.InputVector;
+        }
+
+        /// <summary>
+        /// Re-calculate this <see cref="AgentMovement"/>.
+        /// </summary>
+        /// <param name="change"></param>
+        public void CalculateMovement(ActionDataChanged change)
+        {
+            ActionDataDelta = change.AgentActionData;
+        }
+
+        /// <summary>
+        /// Re-calculate this <see cref="AgentMovement"/>.
+        /// </summary>
+        /// <param name="change"></param>
+        public void CalculateMovement(MountDataChanged change)
+        {
+            MountDataDelta = change.AgentMountData;
+        }
+
+        /// <summary>
+        /// Applies the given <see cref="Agent"/> to the <see cref="AgentData"/>
+        /// returned by this method.
+        /// </summary>
+        /// <param name="agent"></param>
+        /// <returns></returns>
+        public AgentData Apply(Agent agent)
```

Comment:
Move Apply functionality from AgentData to this class. Also don't apply data for default values.